src/deluge/model/song/song.cpp

```diff
@@ -2925,7 +2925,9 @@ void Song::turnSoloingIntoJustPlaying(bool getRidOfArmingToo) {
 			// Just get rid of arming
 			for (int32_t l = 0; l < sessionClips.getNumElements(); l++) {
 				Clip* loopable = sessionClips.getClipAtIndex(l);
-				loopable->armState = ArmState::OFF;
+				if (loopable->launchStyle != LaunchStyle::FILL && loopable->launchStyle != LaunchStyle::ONCE) {
```

Comment:
Should this check that the launch style is default instead?

Reply:
If we ever want to add more launch modes it is better to check for these two specific modes which do something different to normal, this line makes sure they are not unarmed to stop. They are armed before in the code but for Default clips they will get unarmed once they have changed state at the end of the loop. In sum, if we add later another launch mode this code will still work and won't execute unexpected behavior for the new mode which doesn't belong to it. The same applies to other changes in this PR of the 'if' conditions

Reply:
Right but that's the default behaviour, so a new mode wouldn't want to inherit it anyway

Reply:
Oh yea, you are right. Modified it!